L1Trigger/L1TTrackMatch/interface/L1TkMuMantra.h

```diff
@@ -0,0 +1,139 @@
+#ifndef L1TKMUMANTRA_H
+#define L1TKMUMANTRA_H
+
+/*
+** class  : GenericDataFormat
+** author : L.Cadamuro (UF)
+** date   : 4/11/2019
+** brief  : very generic structs to be used as inputs to the correlator
+**        : to make sure that Mantra can handle muons and tracks from all the detectors
+*/
+
+namespace L1TkMuMantraDF {
+
+  struct track_df {
+    double pt;     // GeV
+    double eta;    // rad, -inf / +inf
+    double theta;  // rad, 0 -> +90-90
+    double phi;    // rad, -pi / + pi
+    int nstubs;    //
+    double chi2;   //
+    int charge;    // -1. +1
+  };
+
+  struct muon_df {
+    double pt;     // GeV
+    double eta;    // rad, -inf / +inf
+    double theta;  // rad, 0 -> +90-90
+    double phi;    // rad, -pi / + pi
+    int charge;    // -1. +1
+  };
+}  // namespace L1TkMuMantraDF
+
+/*
+** class  : L1TkMuMantra
+** author : L.Cadamuro (UF)
+** date   : 4/11/2019
+** brief  : correlates muons and tracks using pre-encoded windows
+*/
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <utility>
+#include "L1Trigger/L1TTrackMatch/interface/MuMatchWindow.h"
+// #include "GenericDataFormat.h"
+#include "TFile.h"
+#include "TMath.h"
+
+class L1TkMuMantra {
+public:
+  L1TkMuMantra(std::vector<double>& bounds, TFile* fIn_theta, TFile* fIn_phi, std::string name);
+  ~L1TkMuMantra(){};
+
+  // returns a vector with the same size of muons, each with an index to the matched L1 track, or -1 if no match is found
+  std::vector<int> find_match(std::vector<L1TkMuMantraDF::track_df>& tracks,
+                              std::vector<L1TkMuMantraDF::muon_df>& muons);
+
+  void test(double eta, double pt);
+
+  void relax_windows(double& low, double cent, double& high);  // will modify low and high
+
+  void set_safety_factor(float sf_l, float sf_h) {
+    safety_factor_l_ = sf_l;
+    safety_factor_h_ = sf_h;
+    if (verbosity_ > 0)
+      std::cout << "L1TkMuMantra : " << name_ << " safety factor LOW is " << safety_factor_l_ << std::endl;
+    if (verbosity_ > 0)
+      std::cout << "L1TkMuMantra : " << name_ << " safety factor HIGH is " << safety_factor_h_ << std::endl;
+  }
+
+  int sign(double x) {
+    if (x == 0)
+      return 1;
+    return int(x / std::abs(x));
+  }
+
+  void setArbitrationType(std::string type);  // MaxPt, MinDeltaPt
+
+  // static functions, meant to be used from outside to interface with MAnTra
+  static std::vector<double> prepare_corr_bounds(std::string fname, std::string hname);
+
+  // converters
+  static double deg_to_rad(double x) { return (x * TMath::Pi() / 180.); }
```

Comment:
see above comments about converters
in general, do not copy the same functions in every class (even if custom functions are needed)

Reply:
done